flycheck.el

```diff
@@ -2762,6 +2783,36 @@ because some users override that function, as described in URL
   (flycheck-clear-idle-change-timer)
   (flycheck-buffer-automatically 'idle-change))
 
+(defvar flycheck-buffer-switch-timer nil
+  "Timer to mark the idle time since the last buffer switch.
+This is global so that switching between multiple buffers quickly
+does not trigger a syntax check in each.")
+
+(defun flycheck-clear-buffer-switch-timer ()
+  "Clear the buffer switch timer."
+  (when flycheck-buffer-switch-timer
+    (cancel-timer flycheck-buffer-switch-timer)
+    (setq flycheck-buffer-switch-timer nil)))
+
+(defun flycheck-handle-buffer-switch ()
+  "Handle a possible buffer switch.
+If the buffer was actually switched, trigger a delayed syntax
+check."
+  (unless (eq (current-buffer) (cadr (buffer-list)))
+    (when flycheck-mode
+      (flycheck-clear-buffer-switch-timer)
+      (setq flycheck-buffer-switch-timer
+            (run-at-time flycheck-buffer-switch-delay nil
+                         #'flycheck-handle-idle-buffer-switch
+                         (current-buffer))))))
+
+(defun flycheck-handle-idle-buffer-switch (buffer)
+  "Handle an expired idle timer in BUFFER since the last buffer switch."
+  (when (buffer-live-p buffer)
+    (with-current-buffer buffer
+      (flycheck-clear-idle-change-timer)
```

Comment:
Is this for preventing an automatic check from `idle-change` to be triggered?  It seems to me the interaction of these two timers warrants some discussion. 

What happens if after switching to a buffer, I edit some text, then wait?  With both `idle-change` and `buffer-switch`, surely I should get a check, preferably only one, on `idle-change`, after the `idle-change-delay` has expired.  This means that a change should also clear the `buffer-switch-timer`.